AmplifyPlugins/Auth/AWSCognitoAuthPlugin/Dependency/AuthenticationProviderAdapter+SignIn.swift

```diff
@@ -146,28 +167,49 @@ extension AuthenticationProviderAdapter {
                 DispatchQueue.main.async {
                     navController.dismiss(animated: false) {
                         guard let self = self else { return }
-
-                        if let error = error {
-                            let authError = self.convertSignInUIErrorToAuthError(error)
-                            completionHandler(.failure(authError))
-                            return
-                        }
-
-                        guard let state = state, state == .signedIn else {
-
-                            let error = AuthError.unknown("signInWithWebUI did not produce a valid result.")
-                            completionHandler(.failure(error))
-                            return
-                        }
-                        let authResult = AuthSignInResult(nextStep: .done)
-                        completionHandler(.success(authResult))
+                        self.handleHostedUIResult(state: state, error: error, completionHandler: completionHandler)
                     }
                 }
             }
-
         })
     }
 
+    @available(iOS 13, *)
+    private func launchASWebAuthenticationSession(window: UIWindow,
+                                                  hostedUIOptions: HostedUIOptions,
+                                                  preferPrivateSession: Bool,
+                                                  completionHandler: @escaping SigInResultCompletion) {
+        awsMobileClient.showSignIn(uiwindow: window,
+                                   hostedUIOptions: hostedUIOptions) { [weak self] state, error in
+            guard let self = self else { return }
+            self.handleHostedUIResult(state: state,
+                                      error: error,
+                                      preferPrivateSession: preferPrivateSession,
+                                      completionHandler: completionHandler)
+        }
+    }
+
+    private func handleHostedUIResult(state: UserState?,
+                                      error: Error?,
+                                      preferPrivateSession: Bool = false,
+                                      completionHandler: @escaping SigInResultCompletion) {
+        if let error = error {
+            let authError = convertSignInUIErrorToAuthError(error)
+            completionHandler(.failure(authError))
+            return
+        }
+
+        guard let state = state, state == .signedIn else {
+
+            let error = AuthError.unknown("signInWithWebUI did not produce a valid result.")
```

Comment:
Would be helpful to include the state in the error message if present but not signed in